### PR TITLE
style: Cleanup the cascade a good bit.

### PR DESCRIPTION
**Pull request by emilio**
Was about the time.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17788)

### CODE REVIEW

**Comment by emilio** [comment]
As a followup, we should really get that font-size stuff on Gecko cleaned up...

**Comment by emilio** [comment]
Feel free to just land if I haven't messed up:

https://treeherder.mozilla.org/#/jobs?repo=try&revision=04c9a0924371226288ca7563341ed8dea2b849de

File: components/style/properties/properties.mako.rs [hunk 1]
```diff
@@ -3087,33 +3148,26 @@ where
...
 
     {
-        StyleAdjuster::new(&mut style)
-            .adjust(
-                layout_parent_style,
-                context.device.default_computed_values(),
-                flags
-            );
+        StyleAdjuster::new(&mut builder)
+            .adjust(layout_parent_style, flags);
```
Comment (heycam) [inline components/style/properties/properties.mako.rs]: I guess this doesn't need to be in a block.

File: ports/geckolib/glue.rs [hunk 2]
```diff
@@ -2994,21 +2984,16 @@ pub extern "C" fn Servo_GetComputedKeyframeValues(keyframes: RawGeckoKeyframeLis
...
     let style = ComputedValues::as_arc(&style);
 
-    let element = GeckoElement(element);
-    let parent_element = element.inheritance_parent();
-    let parent_data = parent_element.as_ref().and_then(|e| e.borrow_data());
-    let parent_style = parent_data.as_ref().map(|d| d.styles.primary()).map(|x| &**x);
-
     let pseudo = style.pseudo();
-    let mut context = create_context(&data, &metrics, style, parent_style, pseudo.as_ref());
+    let mut context = create_context(&data, &metrics, style, pseudo.as_ref());
```
Comment (heycam) [inline ports/geckolib/glue.rs]: Are you sure these animation-related function changes are right?  For `Servo_GetAnimationValue`, at least, which is used for SMIL, it seems like Gecko will indeed use the element's parent style if `inherit` was given.

File: ports/geckolib/glue.rs [hunk 3]
```diff
@@ -2908,7 +2903,7 @@ fn create_context<'a>(per_doc_data: &'a PerDocumentStyleDataImpl,
 
 #[no_mangle]
 pub extern "C" fn Servo_GetComputedKeyframeValues(keyframes: RawGeckoKeyframeListBorrowed,
-                                                  element: RawGeckoElementBorrowed,
+                                                  _element: RawGeckoElementBorrowed,
```
Comment (heycam) [inline ports/geckolib/glue.rs]: Remove the argument (maybe in a followup given this means you need to touch C++ too)?

**Review by heycam — COMMENTED** [review]
These changes look good!  Just a question about the animation-related FFI function changes.

File: components/style/properties/properties.mako.rs [hunk 4]
```diff
@@ -2555,6 +2624,15 @@ impl<'a> StyleBuilder<'a> {
         )
     }
 
+    /// Returns the style we're inheriting from.
+    pub fn inherited_style(&self) -> &'a ComputedValues {
+        self.inherited_style
+    }
+
+    /// Returns the style we're getting reset properties from.
+    pub fn reset_style(&self) -> &'a ComputedValues {
```
Comment (emilio) [inline components/style/properties/properties.mako.rs]: Agreed, I used default_style for consistency with computed::Context.

File: components/style/properties/properties.mako.rs [hunk 5]
```diff
@@ -2453,12 +2453,28 @@ impl<'a, T: 'a> ops::Deref for StyleStructRef<'a, T> {
...
+    pub device: &'a Device,
+
+    /// The style we're inheriting from.
+    inherited_style: &'a ComputedValues,
+
+    /// The style we're getting reset structs from.
+    reset_style: &'a ComputedValues,
+
+    /// The style we're inheriting from explicitly, or none if we're the root of
+    /// a subtree.
```
Comment (emilio) [inline components/style/properties/properties.mako.rs]: Will do.

File: ports/geckolib/glue.rs (hunk 2 above)
Comment (emilio) [inline ports/geckolib/glue.rs]: Yeah, this is quite fishy indeed, will fix.

**Pull request merged**